PLUGINS.md

```diff
@@ -0,0 +1,264 @@
+# containerd plugins
+
+containerd supports extending its functionality using most of its defined
+interfaces. This includes using a customized runtime, snapshotter, content
+store, and even adding gRPC interfaces.
+
+## smart client model
+
+containerd has a smart client architecture, meaning any functionality which is
+not required by the daemon is done by the client. This includes most high
+level interactions such as creating a container's specification, interacting
+with an image registry, or loading an image from tar. Using containerd's Go
+client gives a users access to many points of extensions from creating their own
+options on container creation to resolving image registry names.
+
+See [containerd's Go documentation](https://godoc.org/github.com/containerd/containerd)
+
+## external plugins
+
+External plugins allow extending containerd's functionality using an officially
+released version of containerd without recompiling to add a plugin. containerd
+has two methods to use external plugins, either by making a binary available
```

Comment:
```
allows extension through 2 methods:
  - via a binary available in containerd's PATH
  - by configuring containerd as a proxy to another gRPC service
```

?

Reply:
yeah I'll update to a format like that, I think that is clearer than a run-on sentence